src/ty.md

```diff
@@ -2,94 +2,96 @@
-| `rustc_hir::Ty` has generics and lifetimes; however, some of those lifetimes are special markers like [`LifetimeName::Implicit`][implicit]. | `ty::Ty` has the full type, including generics and lifetimes, even if the user left them out |
-| `fn foo(x: u32) → u32 { }` - Two `rustc_hir::Ty` representing each usage of `u32`. Each has its own `Span`s, etc.- `rustc_hir::Ty` doesn’t tell us that both are the same type | `fn foo(x: u32) → u32 { }` - One `ty::Ty` for all instances of `u32` throughout the program.- `ty::Ty` tells us that both usages of `u32` mean the same type. |
-| `fn foo(x: &u32) -> &u32)`- Two `rustc_hir::Ty` again.- Lifetimes for the references show up in the `rustc_hir::Ty`s using a special marker, [`LifetimeName::Implicit`][implicit]. | `fn foo(x: &u32) -> &u32)`- A single `ty::Ty`.- The `ty::Ty` has the hidden lifetime param |
+**Order** 
 
-[implicit]: https://doc.rust-lang.org/nightly/nightly-rustc/rustc_hir/enum.LifetimeName.html#variant.Implicit
+[HIR](hir.md) is built off the [AST](syntax-intro.md) before any [`ty::Ty`] is
+produced. After [HIR](hir.md) some type inference and type checking is done.
+During type inference a [`ty::Ty`] for every element is determined and
+ambiguities are identified. Now [`ty::Ty`] can be checked to confirm typing
+information.
 
-**Order** HIR is built directly from the AST, so it happens before any `ty::Ty` is produced. After
-HIR is built, some basic type inference and type checking is done. During the type inference, we
-figure out what the `ty::Ty` of everything is and we also check if the type of something is
-ambiguous. The `ty::Ty` then, is used for type checking while making sure everything has the
-expected type. The [`astconv` module][astconv] is where the code responsible for converting a
-`rustc_hir::Ty` into a `ty::Ty` is located. This occurs during the type-checking phase,
-but also in other parts of the compiler that want to ask questions like "what argument types does
-this function expect?"
+The [`astconv`][astconv] module handles the actual conversion of a
+[`rustc_hir::Ty`][hir_ty] into a [`ty::Ty`] during the type-checking phase, or
+in other parts of the compiler which need specific type-based information such
+as (but not limited to): "what argument types does this function expect?"
 
-[astconv]: https://doc.rust-lang.org/nightly/nightly-rustc/rustc_typeck/astconv/index.html
+**Semantics of `Ty`** 
 
-**How semantics drive the two instances of `Ty`** You can think of HIR as the perspective
-of the type information that assumes the least. We assume two things are distinct until they are
-proven to be the same thing. In other words, we know less about them, so we should assume less about
-them.
+[HIR](hir.md) assumes the least about type information. Since
+[`rustc_hir::Ty`](hir_ty) knows very little about type distinction it assumes
+much less about types than [`ty::Ty`] which is able to prove indistinction and
+make specific assumptions.
```

Comment:
nit: maybe we should remove "and make specific assumptions" unless we have an example. I personally like the way it was previously written that `hir::Ty` assumes that two types are distinct until proven otherwise.